src/CommunityToolkit.Maui/HandlerImplementation/Popup/Popup.macios.cs

```diff
@@ -1,13 +1,92 @@
-﻿using CommunityToolkit.Maui.Core;
+﻿using System.ComponentModel;
+using CommunityToolkit.Maui.Core;
 using CommunityToolkit.Maui.Core.Handlers;
+using CommunityToolkit.Maui.Core.Views;
 using Microsoft.Maui.Controls;
 using Microsoft.Maui.Handlers;
 using Microsoft.Maui.Platform;
-
+using UIKit;
 namespace CommunityToolkit.Maui.Views;
 
-public partial class Popup
+public partial class Popup : Element
```

Comment:
why this's using the shared layer instead of the Handler and the `MapSize`?

Reply:
@pictos , Since it was unclear how to detect the size change event of the child element of Popup's Content, I implemented it in the shared layer.
If it is possible to detect changes in the size of child elements as the text changes, there is no need to implement it in the shared layer.
Is there any better way?

At least if I didn't implement it like this, No. 4 wouldn't work.
In the case where the Popup's size is unspecified and it resizes according to the dynamic resizing of Content's child elements.

Reply:
@pictos Just curious - what's the downside to making `Popup` inherit from `Element`. 

Looking at the .NET MAUI repo, every control that appears on the screen inherits from `VisualElement` (e.g. `Button`, `Label`, etc, all inherit from `VisualElement`).

For `Popup`, I'd say it may make even more sense to have it inherit from `VisualElement` to better align with how .NET MAUI creates controls.

Reply:
@brminnick answering your questions...
> @pictos Just curious - what's the downside to making Popup inherit from Element. 

All the downside possibles, hehe... The .NET MAUI team gave more love to `VisualElement` instead of `Element`, the good part from the community is that Shane made a lot of improvements on `Element` because of poups, and now everyone in the community will have a happy place where they need to implement controls that inherit from `Element`.

> For Popup, I'd say it may make even more sense to have it inherit from VisualElement to better align with how .NET MAUI creates controls.

Would be great if we could do it, but we are blocked on that because of the current .NET MAUI restrictions on what can be a `VisualElement`. So, every control that inherits from `VisualElement` its native part will inherit from `ViewHandler`, Basically just controls that are inherited from the [platform View](https://github.com/dotnet/maui/blob/main/src/Core/src/Handlers/View/ViewHandlerOfT.cs#L2-L11) can be a `VisualElement`. Our popup implementation follows what each platform defines what's a popup, For android it's a Dialog and for iOS a UIViewController is used, and none of those inherit from _platform View_, with that, we can't make the PopupHandler a `ViewHandler`  and because of that the Popup can't be a `VisualElement`.

Reply:
Got it! Thanks for the details!!

Reply:
@pictos , Thank you for suggesting the amendment.
Fixed to call SetSize method in Content.SizeChanged.